### PR TITLE
add javascript and livescript support

### LINKED ISSUE #69: LiveScript support
LiveScript is a more actively developed CoffeeScript-like language that includes pragmatic, dynamically-typed functional concepts (without becoming academic), as well as better ES6 compatibility. Overall it seems like the CoffeeScript idea thought all the way to the end. Mycha should support it.

* determine if LiveScript or CoffeeScript is used by inspecting the local `package.json` file
* generate a `mycha.ls` file instead of `mycha.coffee` and use it when it exists
* add `'ls:livescript'` to the `compilers` section
* add `.ls` to the testFilePattern section
* in the LiveScript files, make camelCased identifiers spinal-cased

### PR DESCRIPTION
**Pull request by charlierudolph**
@alexdavid @kevgo 

Converted the tests to js since that is the base case. 
Restructured coffee-script support and added livescript support.
Tried to make it easy to add support for other languages

resolves #69

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -24,30 +26,30 @@ and provides a utility that gets tests up and running quickly
...
-    sinon = require 'sinon'
-    chai.use require 'sinon-chai'
+    ```js
+    var chai = require('chai');
+    var sinon = require('sinon');
+    chai.use(require('sinon-chai'));
 
-    global.expect = chai.expect
-    global.sinon = sinon
+    global.chai = chai;
```
Comment (kevgo) [inline README.md]: What's the global `chai` for? Afaik it isn't used directly.

File: README.md [hunk 2]
```diff
@@ -59,6 +61,11 @@ and provides a utility that gets tests up and running quickly
   * if folders or files are specified, pass them through to mocha
 
 
+## Compilers
+
+To add a new compiler, add an entry [here](src/compilers.coffee) and add a
+language specific version of the test helper [here](scaffold)
```
Comment (kevgo) [inline README.md]: Add a period at the end of this sentence.

File: features/install.feature [hunk 3]
```diff
@@ -5,7 +5,45 @@ Feature: Installation
   So that I can start testing without repetitive setup work.
 
 
-  Scenario: fresh install
+  Scenario: install with a .js test helper
+    Given I am working on a Node.js project
+    When I run "mycha install" and enter "spec/spec_helper.js"
+    Then my "package.json" now lists the devDependencies
```
Comment (kevgo) [inline features/install.feature]: Unless you overwrite the devDependencies, this step should say 
> Then my "package.json" now lists the __additional__ devDependencies

File: features/install.feature [hunk 4]
```diff
@@ -5,7 +5,45 @@ Feature: Installation
...
+      """
+      var chai = require('chai');
+      var sinon = require('sinon');
+      chai.use(require('sinon-chai'));
+
+      global.chai = chai;
+      global.expect = chai.expect;
+      global.sinon = sinon;
+
+      process.env.NODE_ENV = 'test';
```
Comment (kevgo) [inline features/install.feature]: I think its best to move this to set this as early as possible, i.e. move this to the first line in the file.

File: features/install.feature [hunk 5]
```diff
@@ -5,7 +5,45 @@ Feature: Installation
   So that I can start testing without repetitive setup work.
 
 
-  Scenario: fresh install
+  Scenario: install with a .js test helper
+    Given I am working on a Node.js project
+    When I run "mycha install" and enter "spec/spec_helper.js"
```
Comment (kevgo) [inline features/install.feature]: Is it really the best user experience to have to enter the full path of this file? Maybe its better to show some sort of prompt using https://github.com/sboudrias/Inquirer.js?

File: features/install.feature [hunk 6]
```diff
@@ -37,9 +75,47 @@ Feature: Installation
         reporter: dot
 
       # Path patten used for finding tests (see https://github.com/isaacs/minimatch)
-      testFilePattern: '**/*_{spec,test}.{coffee,js}'
+      testFilePattern: '**/*_{spec,test}.coffee'
```
Comment (kevgo) [inline features/install.feature]: Since dashes are valid in filenames as well, and others might use camelCase filenames, we should get rid of the `_` here: `'**/*{spec,test}.coffee'`

File: features/install.feature [hunk 7]
```diff
@@ -37,9 +75,47 @@ Feature: Installation
...
+      process.env.NODE_ENV = 'test'
+      """
+    And my project now has a file "mycha.yml" containing
+      """
+      # Environment variables to add to process.env when running mocha
+      mochaEnv: {}
+
+      # Default options to pass to mocha (can be overriden by command line options)
+      mochaOptions:
+        colors: true
```
Comment (kevgo) [inline features/install.feature]: I think `colors: yes` reads a bit nicer.

Comment (kevgo) [inline features/install.feature]: I think `colors: yes` reads a bit better, and this is supported by YML.

File: src/compilers.coffee [hunk 8]
```diff
@@ -0,0 +1,8 @@
+# key is the file extension
+# value is the module name
+
+# See the --compilers documentation at http://mochajs.org/
+
+module.exports =
+  'coffee': 'coffee-script/register'
```
Comment (kevgo) [inline src/compilers.coffee]: Keys don't have to be strings.

```coffeescript
coffee: 'coffee-script/register'
ls: 'livescript

File: features/install.feature (hunk 5 above)
Comment (charlierudolph) [inline features/install.feature]: On asf we did specs inline with source, and the spec folder was just for helpers.

File: features/install.feature (hunk 7 above)
Comment (charlierudolph) [inline features/install.feature]: Thats not in the yaml spec http://www.yaml.org/spec/1.2/spec.html 
and isn't support by js-yaml http://nodeca.github.io/js-yaml/#yaml=eDogeWVz

File: src/compilers.coffee (hunk 8 above)
Comment (charlierudolph) [inline src/compilers.coffee]: :+1:

File: features/install.feature (hunk 6 above)
Comment (charlierudolph) [inline features/install.feature]: sounds good

File: features/install.feature (hunk 4 above)
Comment (charlierudolph) [inline features/install.feature]: sounds good

File: features/install.feature (hunk 3 above)
Comment (charlierudolph) [inline features/install.feature]: k

File: README.md (hunk 2 above)
Comment (charlierudolph) [inline README.md]: k

File: README.md (hunk 1 above)
Comment (charlierudolph) [inline README.md]: I've seen us use it fo  other helper files which add custom matchers, but that can also be accomplished by another call to `chai.use` in this file. I'll remove it.

File: features/install.feature (hunk 3 above)
Comment (charlierudolph) [inline features/install.feature]: Don't really like the sound of that going with 
```cucumber
Then my "package.json" devDependencies now includes
```

File: README.md (hunk 1 above)
Comment (alexdavid) [inline README.md]: You were the one who added this :smile: 

https://github.com/Originate/mycha/commit/10f2883b10d67f173c7503508be94200e429d10c

File: features/install.feature (hunk 5 above)
Comment (charlierudolph) [inline features/install.feature]: This is also what we had previously. If we want to change the prompt can that be dealt with in another PR?

Comment (kevgo) [inline features/install.feature]: Yes, we can changes this separately. I guess the path is just there the spec_helper file is to be created then.

File: features/install.feature (hunk 7 above)
Comment (kevgo) [inline features/install.feature]: Okay. Sorry I only tested it with http://yaml-online-parser.appspot.com, where this works, and taken the idea from http://yaml.org/type/bool.html. Strange.

File: README.md (hunk 1 above)
Comment (kevgo) [inline README.md]: Haha, sounds like I was ahead of myself here there. Let's leave it on.

File: features/install.feature (hunk 7 above)
Comment (charlierudolph) [inline features/install.feature]: Looks like it was in yaml 1.1, but removed in yaml 1.2

Comment (kevgo) [inline features/install.feature]: They couldn't leave it alone...  :-/
